### PR TITLE
BaSD-MS-58: Reduce Applications

### PR DESCRIPTION
**Pull request by ripellizzari**
**### Refactor made taking into account what's been done on admins entity.**

**Redux Global States implemented.**

--------

Finished: 
-Reduce getApplications
-Reduce deleteApplications
-Refactor Applications.js
-Refactor Application.js
-Reduce addApplications
-Reduce updateApplications
-Refactor Form
-Dispatch data from Position, Client and Postulant.

### CODE REVIEW

File: src/Components/Applications/Form/index.js [hunk 1]
```diff
@@ -5,175 +5,75 @@ import Select from '../../Shared/Select';
...
-        );
-        setClientValue(res[0]._id);
-      })
-      .catch((error) => setError(error.toString()))
-      .finally(() => {
-        setLoading((prev) => {
-          return { ...prev, clientLoading: false };
-        });
-      });
+  const values = {
```
Comment (nicolasviola) [inline src/Components/Applications/Form/index.js:54]: Esta constante deberia ser creada dentro de la funcion "onSubmit"

File: src/redux/applications/reducer.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    }
+    case GET_APPLICATIONS_OPTIONS_FETCHING:
+      return {
+        ...state,
+        isLoading: true
+      };
+    case GET_APPLICATIONS_OPTIONS_FULFILLED: {
+      const addedOptions = action.payload.map((option) => {
+        let lebelValue = '';
+        if (option.firstName !== undefined) {
```
Comment (nicolasviola) [inline src/redux/applications/reducer.js:142]: Podes hacer directamente if(!option.firstName)

**Review by nicolasviola — CHANGES_REQUESTED** [review]

File: src/Components/Applications/Form/index.js [hunk 3]
```diff
@@ -91,7 +95,7 @@ function ProfilesForm() {
           setPostulantsValue(data?.postulants?._id);
           setResult(data.result);
         })
-        .catch((error) => setError(error.toString()))
+        .catch((error) => error.toString())
         .finally(() =>
```
[the remark this answers is not in the document]
Comment (ripellizzari) [reply src/Components/Applications/Form/index.js]: Ya fue modificado el codigo.

File: src/Components/Applications/Form/index.js (hunk 1 above)
Comment (ripellizzari) [reply src/Components/Applications/Form/index.js:54]: hecho!

File: src/redux/applications/reducer.js (hunk 2 above)
Comment (ripellizzari) [reply src/redux/applications/reducer.js:142]: Probe de esta manera y me devolvio undefined

**Review by nicolasviola — APPROVED** [review]

**Pull request merged**